### ISSUE TITLE
"Your Google tag configuration has changed" banner appears behind the gathering data / zero data state banners

### ISSUE THREAD

**Issue by bethanylang** [google/site-kit-wp#6766]
## Bug Description

This issue was discovered in Bug Bash by @wpdarren.

"Your Google tag configuration has changed" banner only appears after the gathering data or zero data notifications have been closed down.

## Steps to reproduce

1. Make sure that the GTE Support feature flag is enabled. Setup Site Kit and connect a GA4 Property. 
2. For easy testing purposes, go to Site Kit settings and click `Edit in Analytics` next to the Google Analytics 4 Property and rename it to say `GA4 Property `1. Also, in SiteKit settings, note the `Measurement ID` of `GA4 Property 1`.
3. Go to [Tag Manager admin](https://tagmanager.google.com/) and sign in using the same Google Account which has access to the above `GA4 Property 1`. 
4. On the `Google tags` tab, find the row where one of the Tag IDs is the same as the `Measurement ID` above. The Tag Name would by default be the domain name associated to the GA4 Property. 
5. Click on this tag name. On the Tag details pop-up, note down the tag ID beginning with GT (say GT-100000). 
6. Delete the destination (`GA4 Property 1`) from the Destinations list. When asked to attach a tag to the destination, create a new tag. This will make the original GT-100000 tag not linked to any destination.
7. Now go back to the Site Kit plugin and in the browser console, run the following command which forces the Google Tag settings to sync.

`googlesitekit.api.set('modules', 'analytics-4', 'settings', {'googleTagLastSyncedAtMs': 0});`

8. Reload the plugin and see the issue as reported. 

Please note as per the AC on [this ticket](https://github.com/google/site-kit-wp/issues/6373) there are two slightly different Banner notification scenarios, which you might need to be aware of. 

## Screenshots

![image](https://user-images.githubusercontent.com/12189958/227261450-383760e1-c3f5-434f-8d9f-96fb67f30e07.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

Note that the present priority order of notifications is as follows:
1. Setup success notifications.
2. Enable auto-update notification.
3. Core site notifications.
4. Module recovery alert.
5. GA4 activation banner.
6. Zero state notifications (gathering data & zero data).
7. Key metrics "Personalize your metrics" notification.
8. AdSense alerts.
9. Google tag ID mismatch notification (the "Your Google tag configuration has changed" banner).
10. Switch to GA4 dashboard view notification.

---

* The "Your Google tag configuration has changed" banner that was implemented in #6373 should appear directly above the zero state notifications in terms of priority.
* In other words, it should appear after the notifications up to and including the GA4 activation banner, and before any of the others if they are both present on a given page load.

## Implementation Brief

* In `assets/js/components/notifications/BannerNotifications.js`:
    * Move the render of `GoogleTagIDMismatchNotification` from the `Fragment` below the `ZeroDataStateNotifications` to the `Fragment` above it, directly below the `ActivationBanner` element. 

### Test Coverage

* Fix any failing tests due to the change. 

## QA Brief

* Follow the step to reproduce. 
* The reported issue should be fixed and the GTE banner should display. 

## Changelog entry

* Fix placement order of Google Tag banner notification.

**Comment by aaemnnosttv** [comment]
Looks like the priority of this notification was never specified in #6083 or #6373 so it was added to the bottom. Should be a quick one to address.

**Comment by techanvil** [comment]
Sounds good, thanks @marrrmarrr!

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

@kuasha420 

- I've noticed that if zero data banner exist then before loading the "Your Google tag configured has changed" banner for few seconds Zero or gathering state banner appears before loader.

- When we refresh the page, "Your Google tag configured has changed" banner gets hide. 

Are both cases expected ?


https://user-images.githubusercontent.com/94359491/229474273-fa580016-c3ce-40d3-8d94-d99092f019ee.mp4

**Comment by mohitwp** [comment]
### QA Update ✅ 

Thank you @kuasha420 @techanvil 

- Tested on dev.
- "Your Google tag configuration has changed" banner now appearing over the zero/gathering data banners.

- I created follow-up issue https://github.com/google/site-kit-wp/issues/6819 to address issue reported [above](https://github.com/google/site-kit-wp/issues/6766#issuecomment-1494016786). 

<details>

https://user-images.githubusercontent.com/94359491/229474273-fa580016-c3ce-40d3-8d94-d99092f019ee.mp4

</details>
